packages/core/src/behavior.ts

```diff
@@ -135,51 +134,54 @@ export function createPromiseBehavior<T, TEvent extends EventObject>(
   const observers: Set<Observer<T>> = new Set();
```

Comment:
if I recall this correctly - behaviors in Akka are pure, right? is the goal to make them pure in XState? if they are not pure - is calling them behaviors appropriate?

Reply:
Yes, they should be pure. To make them pure, they would look more like reducers than closures (which is what's happening in some of the behavior implementations)... if we want to make them pure, we have to ensure that internal state (such as the `canceled` variable here) isn't exposed as an emitted value. This might mean changing the typings.

Reply:
A somewhat bigger challenge here, than typings, would be how we should keep around and expose the public and internal "snapshots" - having access to the internal state is crucial for persisting & rehydrating

Reply:
That is why there is the potential for `.getSnapshot()` to be different than what `TEmitted` represents. This is why we need two types.